server/src/weaverbird/pipeline/steps/utils/combination.py

```diff
@@ -55,10 +55,12 @@ def _pipelinestep_adapter() -> TypeAdapter["str | list[PipelineStep]"]:
 
 
 @cache
-def _pipelinestepwithref_adapter() -> TypeAdapter["str | list[PipelineStepWithRefs | PipelineStep]"]:
-    from weaverbird.pipeline.pipeline import PipelineStep, PipelineStepWithRefs
+def _pipelinestepwithref_adapter() -> (
+    TypeAdapter["str | list[PipelineStepWithRefs | PipelineStepWithVariables | PipelineStep]"]
+):
+    from weaverbird.pipeline.pipeline import PipelineStep, PipelineStepWithRefs, PipelineStepWithVariables
 
-    return TypeAdapter(str | list[PipelineStepWithRefs | PipelineStep])  # type: ignore[arg-type]
+    return TypeAdapter(str | list[PipelineStepWithRefs | PipelineStep | PipelineStepWithVariables])  # type: ignore[arg-type]
```

Comment:
Could you please add a comment indicating that `PipelineStepWithVariables` must be last, otherwise we might end up with `WithVariables` variants rather than regular `PipelineStep`s ?

Reply:
Ah yes ideed !

Reply:
What do you think of:
```py
# Note: PipelineStep must appear _before_ PipelineStepWithVariables, to avoid fields that could contain variables
# to always be matched as strings
```